src/OneDriveAuth.js

```diff
@@ -133,7 +133,7 @@ export class OneDriveAuth {
           cachePlugin,
         };
       }
-      this._app = this.onCode
+      this._app = this.acquireMethod === AcquireMethod.BY_DEVICE_CODE
         ? new PublicClientApplication(msalConfig)
         : new ConfidentialClientApplication(msalConfig);
     }
```

Comment:
Hm, I guess this is identical, but probably better readable

Reply:
no, the presence of the `onCode` callback should not influence the app to be used.